### PR TITLE
EIP1-2386 Process batch responses file with MySQL

### PR DESCRIPTION
**Pull request by Neil-Massey**
Adds processing of response file using the MySQL database records.

Subsequent PR will remove the Dynamo interactions.

### CODE REVIEW

File: src/test/kotlin/uk/gov/dluhc/printapi/service/PrintResponseProcessingServiceTest.kt [hunk 1]
```diff
@@ -166,6 +174,80 @@ class PrintResponseProcessingServiceTest {
             assertThat(captor.allValues[1]).usingRecursiveComparison().isEqualTo(listOf(expectedPrintDetails2))
             verify(idFactory).requestId()
         }
+
+        @Test
+        fun `should update all certificates within each batch`() {
```
Comment (Neil-Massey) [inline src/test/kotlin/uk/gov/dluhc/printapi/service/PrintResponseProcessingServiceTest.kt:179]: Mirrored existing test, but would probably prefer this as separate tests for `failed batch` and `success batch` and possibly `multiple batches`

File: src/test/kotlin/uk/gov/dluhc/printapi/service/PrintResponseProcessingServiceTest.kt [hunk 2]
```diff
@@ -166,6 +174,80 @@ class PrintResponseProcessingServiceTest {
...
+            service.processBatchResponses(listOf(batchResponse1, batchResponse2))
+
+            // Then
+            verify(certificateRepository).findByStatusAndPrintRequestsBatchId(SENT_TO_PRINT_PROVIDER, batchResponse1.batchId)
+            verify(certificateRepository).findByStatusAndPrintRequestsBatchId(SENT_TO_PRINT_PROVIDER, batchResponse2.batchId)
+            verify(certificateRepository).saveAll(listOf(certificate1))
+            verify(certificateRepository).saveAll(listOf(certificate2))
+            assertThat(certificate1.getCurrentPrintRequest().requestId).isEqualTo(newRequestId)
+            assertThat(certificate1.getCurrentPrintRequest().batchId).isNull()
+            assertThat(
```
Comment (Neil-Massey) [inline src/test/kotlin/uk/gov/dluhc/printapi/service/PrintResponseProcessingServiceTest.kt:227]: As this is not a `data` class no option to use the `copy` approach used in the previous DynamoDb tests.

File: src/main/kotlin/uk/gov/dluhc/printapi/service/PrintResponseProcessingService.kt [hunk 3]
```diff
@@ -71,8 +83,66 @@ class PrintResponseProcessingService(
...
+                if (status != SUCCESS) {
+                    val currentPrintRequest = it.getCurrentPrintRequest()
+                    currentPrintRequest.requestId = idFactory.requestId()
+                    currentPrintRequest.batchId = null
+                }
+            }
+        }
+    }
+
     fun processPrintResponse(printResponse: ProcessPrintResponseMessage) {
```
Comment (NathanRussellValtech) [inline src/main/kotlin/uk/gov/dluhc/printapi/service/PrintResponseProcessingService.kt:119]: Does this method need to be `@Transactional`?

**Review by NathanRussellValtech — APPROVED** [review]
LGTM; just one minor comment about whether the method should be `@Transactional` ?
Other than that 👍

File: src/main/kotlin/uk/gov/dluhc/printapi/service/PrintResponseProcessingService.kt (hunk 3 above)
Comment (Neil-Massey) [reply src/main/kotlin/uk/gov/dluhc/printapi/service/PrintResponseProcessingService.kt:120]: Yes. definitely - thanks for spotting 👍

**Pull request merged**